akka-stream/src/main/scala/akka/stream/ActorFlowMaterializer.scala

```diff
@@ -35,11 +31,11 @@ object ActorFlowMaterializer {
    * the processing steps. The default `namePrefix` is `"flow"`. The actor names are built up of
    * `namePrefix-flowNumber-flowStepNumber-stepName`.
    */
-  def apply(materializerSettings: Option[ActorFlowMaterializerSettings] = None, namePrefix: Option[String] = None)(implicit context: ActorRefFactory): ActorFlowMaterializer = {
+  def apply(materializerSettings: Option[ActorFlowMaterializerSettings] = None, namePrefix: Option[String] = None, optimizations: Optimizations = Optimizations.none)(implicit context: ActorRefFactory): ActorFlowMaterializer = {
```

Comment:
optimizations are ignored for now. In fact I am no longer sure that the optimizer should be here. It is much better to move this to a separate optimizer so you don't need to reoptimize every time the same stuff.